monai/metrics/meandice.py

```diff
@@ -9,231 +9,115 @@
-from typing import Callable, Optional, Union
+from typing import Optional, Union
 
 import torch
 
-from monai.networks import one_hot
-from monai.utils import MetricReduction
+from monai.metrics.utils import *
 
 
 class DiceMetric:
     """
     Compute average Dice loss between two tensors. It can support both multi-classes and multi-labels tasks.
-    Input logits `y_pred` (BNHW[D] where N is number of classes) is compared with ground truth `y` (BNHW[D]).
-    Axis N of `y_preds` is expected to have logit predictions for each class rather than being image channels,
-    while the same axis of `y` can be 1 or N (one-hot format). The `include_background` class attribute can be
-    set to False for an instance of DiceLoss to exclude the first category (channel index 0) which is by
-    convention assumed to be background. If the non-background segmentations are small compared to the total
-    image size they can get overwhelmed by the signal from the background so excluding it in such cases helps
-    convergence.
+    Input `y_pred` (BNHW[D] where N is number of classes) is compared with ground truth `y` (BNHW[D]).
+    `y_preds` is expected to have binarized predictions and `y` should be in one-hot format. You can use suitable transforms
+    in ``monai.transforms.post`` first to achieve binarized values.
+    The `include_background` parameter can be set to ``False`` for an instance of DiceLoss to exclude
+    the first category (channel index 0) which is by convention assumed to be background. If the non-background
+    segmentations are small compared to the total image size they can get overwhelmed by the signal from the
+    background so excluding it in such cases helps convergence.
 
     Args:
         include_background: whether to skip Dice computation on the first channel of
             the predicted output. Defaults to True.
-        to_onehot_y: whether to convert `y` into the one-hot format. Defaults to False.
-        mutually_exclusive: if True, `y_pred` will be converted into a binary matrix using
-            a combination of argmax and to_onehot.  Defaults to False.
-        sigmoid: whether to add sigmoid function to y_pred before computation. Defaults to False.
-        other_act: callable function to replace `sigmoid` as activation layer if needed, Defaults to ``None``.
-            for example: `other_act = torch.tanh`.
-        logit_thresh: the threshold value used to convert (for example, after sigmoid if `sigmoid=True`)
-            `y_pred` into a binary matrix. Defaults to 0.5.
         reduction: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
             ``"mean_channel"``, ``"sum_channel"``}
             Define the mode to reduce computation result of 1 batch data. Defaults to ``"mean"``.
 
-    Raises:
-        ValueError: When ``sigmoid=True`` and ``other_act is not None``. Incompatible values.
-
     """
 
     def __init__(
         self,
         include_background: bool = True,
```

Comment:
How about we have an option `ignore_nans: bool = True`: when it's True, the `__call__` will return the metric (with `nans` replaced by 0) and the result of `non_nans`; when it's False, the `__call__` will directly return the metric without any `nan` replacement.

this is to avoid `self.not_nans = not_nans`, this assignment won't work in a multi-process context